src/ui/public/visualize/visualize.js

```diff
@@ -147,7 +147,10 @@ uiModules
 
               return resp;
             });
-        }, 100);
+        };
+
+        $scope.fetch = _.debounce(fetch, 100);
+        courier.setSearchCallback(fetch);
```

Comment:
As far as I understand this will only set a single callback to be executed when the update timer fires? That won't work imho, since the `<visualize>` can exist several times (on a dashboard), and it could also exist besides some saved searches. Doesn't ALL those need to have their fetch triggered? So wouldn't we need a Set of listeners here instead of a single one?